pyodide_build/cli/xbuildenv.py

```diff
@@ -175,40 +180,103 @@ def _search(
         )
         raise typer.Exit(1)
 
-    table = []
-    columns = [
-        # column name, width
-        ("Version", 10),
-        ("Python", 10),
-        ("Emscripten", 10),
-        ("pyodide-build", 25),
-        ("Compatible", 10),
-    ]
-    header = [f"{name:{width}}" for name, width in columns]
-    divider = ["-" * width for _, width in columns]
-
-    table.append("\t".join(header))
-    table.append("\t".join(divider))
-
-    for release in releases:
-        compatible = (
-            "Yes"
-            if release.is_compatible(
-                python_version=local["python"],
-                pyodide_build_version=local["pyodide-build"],
-            )
-            else "No"
-        )
-        pyodide_build_range = f"{release.min_pyodide_build_version or ''} - {release.max_pyodide_build_version or ''}"
-
-        row = [
-            f"{release.version:{columns[0][1]}}",
-            f"{release.python_version:{columns[1][1]}}",
-            f"{release.emscripten_version:{columns[2][1]}}",
-            f"{pyodide_build_range:{columns[3][1]}}",
-            f"{compatible:{columns[4][1]}}",
+    def _generate_json_output(releases, local) -> str:
```

Comment:
How about extracting out `_generate_json_output`, `_print_table_output` into a class and put those function as a method. So we can make the function smaller and make it more testable. Such as:

```python
@dataclass
class MetadataView:
  version: ...
  python: ...
  emscripten: ...
  pyodide_build: ...
  compatible: ...

  def to_json():
        ...

  def to_table():
        ...
```